tree.rb

```diff
@@ -0,0 +1,14 @@
+def file_tree(item, counter)
+  Dir.glob("#{item}/*").each do |filename|
+    if File.file?(filename)
+      puts "|   "*counter + "|---" + File.basename(filename)
+    elsif File.directory?(filename)
+      puts "|   "*counter + "|---" + File.basename(filename)
+      counter += 1
+      file_tree(filename, counter)
```

Comment:
Instead of incrementing and decrementing counter like that just make your recursive call like this

`file_tree(filename, counter + 1)`